File: backend/backend/app/schemas/record.py
"""Record schemas — frozen by contract (RecordOut, RecordsPage)."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class RecordOut(BaseModel):
    id: uuid.UUID
    dataset_id: uuid.UUID
    sector: str
    data: dict
    recorded_at: datetime | None = None
    created_at: datetime


class RecordsPage(BaseModel):
    sector: str
    total: int
    limit: int
    offset: int
    items: list[RecordOut]

File: frontend/src/types/record.ts
// Matches RecordOut / RecordsPage in openapi.json.

export interface RecordOut {
  id: string;
  dataset_id: string;
  sector: string;
  data: Record<string, unknown>;
  recorded_at?: string | null;
  created_at: string;
}

export interface RecordsPage {
  sector: string;
  total: number;
  limit: number;
  offset: number;
  items: RecordOut[];
}
